ontotext-yasgui-web-component/src/plugins/yasr/charts-plugin.ts

```diff
@@ -0,0 +1,292 @@
+      this.initEditor();
+      this.doDraw();
+    }
+  }
+
+  private doDraw() {
+    this.yasr.resultsEl.innerHTML = '';
+    this.addChartConfigButton();
+
+    // @ts-ignore
+    const dataTable = new google.visualization.DataTable();
+    const jsonResults = this.yasr.results.getAsJson();
+    jsonResults.head.vars.forEach((variable) => {
+      let type = 'string';
+      try {
+        type = this.getGoogleTypeForBindings(jsonResults.results.bindings, variable);
+      } catch (e) {
+        if (e instanceof TypesMappingError) {
+          // yasr.warn(e.toHtml())
+        } else {
+          throw e;
+        }
+      }
+      dataTable.addColumn(type, variable);
+    });
+
+    let usedPrefixes = this.yasr.getPrefixes();
+
+    jsonResults.results.bindings.forEach((binding) => {
+      const row = [];
+      jsonResults.head.vars.forEach((variable, columnId) => {
+        row.push(this.castGoogleType(binding[variable], usedPrefixes, dataTable.getColumnType(columnId)));
+      });
+      dataTable.addRow(row);
+    });
+
+    const pluginHtml = document.createElement("div");
+    pluginHtml.id = 'visualization';
+    this.yasr.resultsEl.appendChild(pluginHtml);
+
+    // @ts-ignore
+    this.wrapper = new google.visualization.ChartWrapper({
+      chartType: 'Table',
+      dataTable: dataTable,
+      containerId: 'visualization'
+    });
+
+    this.wrapper.setOption("width", '100%');
+    this.wrapper.setOption("height", 600);
+    this.wrapper.draw();
+    // this.yasr.updateHeader();
+  }
+
+  getIcon(): Element | undefined {
+    const icon = document.createElement('div');
+    icon.innerHTML = SvgUtil.getYasrChartPluginIcon();
+    return icon;
+  }
+
+  destroy(): void {
+    // TODO remove all listeners if any.
+  }
+
+  private redrawChart() {
+    this.chartEditor.getChartWrapper().draw(document.getElementById('visualization'));
+  }
+
+  private chartLoadHandler() {
+    console.log('loaded',);
+    // @ts-ignore
+    google.charts.load('current', {packages: ['charteditor']}).then(this.onLoadCallback.bind(this));
+    // @ts-ignore
+    // google.charts.setOnLoadCallback(this.onLoadCallback.bind(this));
+  }
+
+  private onLoadCallback() {
+    this.initEditor();
+
+    this.doDraw();
+
+    // chart loader and modules are loaded
+    this.loaded = true;
+  }
+
+  private initEditor() {
+    // @ts-ignore
+    this.chartEditor = new google.visualization.ChartEditor();
+    // @ts-ignore
+    google.visualization.events.addListener(this.chartEditor, 'ok', this.redrawChart.bind(this));
+  }
+
+  private addChartConfigButton() {
+    if (this.yasr.resultsEl.parentElement.querySelector('#openChartConfigBtn')) {
+      return;
+    }
+    const openConfigButton = document.createElement('button');
+    openConfigButton.id = 'openChartConfigBtn';
+    openConfigButton.innerHTML = this.translationService.translate("yasr.plugin_control.plugin.charts.config.button");
+    openConfigButton.addEventListener('click', () => {
```

Comment:
We have to unregister the listeners in destroy function.

Reply:
this exactly is safe imo, but I unregistered the another one